File: daeun/dfs_bfs/타겟 넘버.js
function solution(numbers, target) {
  let answer = 0;

  function dfs(index, sum) {
    // 모든 숫자를 탐색했을 때
    if (index === numbers.length) {
      if (sum === target) {
        answer++;
      }
      return;
    }

    // 현재 숫자를 더하는 경우
    dfs(index + 1, sum + numbers[index]);
    // 현재 숫자를 빼는 경우
    dfs(index + 1, sum - numbers[index]);
  }

  // DFS 탐색 시작
  dfs(0, 0);

  return answer;
}

File: Grace/dfs_bfs/타겟 넘버.py
def solution(numbers, target):
    leaves = [0] 
    answer = 0 # 카운트
    
    for num in numbers:
        temp = []
        
        for leaf in leaves:
            temp.append(leaf + num) 
            temp.append(leaf - num)
        
        leaves = temp
        
    # 타겟과 같은지를 확인
    for leaf in leaves:
        if leaf == target:
            answer += 1
    
    return answer